src/station.rs

```diff
@@ -1,112 +1,102 @@
-    /// Station bssid (u8)
-    pub bssid: Option<Vec<u8>>,
-    /// Time since the station is last connected in seconds (u32)
-    pub connected_time: Option<Vec<u8>>,
-    /// Reception bitrate (u8)
-    pub rx_bitrate: Option<Vec<u8>>,
-    /// Total received packets (MSDUs and MMPDUs) from this station (u32)
-    pub rx_packets: Option<Vec<u8>>,
-    /// Signal strength of last received PPDU (u8, dBm)
-    pub signal: Option<Vec<u8>>,
-    /// Transmission bitrate (u8)
-    pub tx_bitrate: Option<Vec<u8>>,
-    /// Total failed packets (MPDUs) to this station (u32)
-    pub tx_failed: Option<Vec<u8>>,
-    /// Total transmitted packets (MSDUs and MMPDUs) to this station (u32)
-    pub tx_packets: Option<Vec<u8>>,
-    /// Total retries (MPDUs) to this station (u32)
-    pub tx_retries: Option<Vec<u8>>,
+    /// Signal strength average
+    pub average_signal: Option<i8>,
+    /// Count of times beacon loss was detected
+    pub beacon_loss: Option<u32>,
+    /// Station bssid
+    pub bssid: Option<MacAddr>,
+    /// Time since the station is last connected in seconds
+    pub connected_time: Option<u32>,
+    /// Reception bitrate (u32)
+    pub rx_bitrate: Option<u32>,
+    /// Total received packets (MSDUs and MMPDUs) from this station
+    pub rx_packets: Option<u32>,
+    /// Signal strength of last received PPDU
+    pub signal: Option<i8>,
+    /// Transmission bitrate
+    pub tx_bitrate: Option<u32>,
+    /// Total failed packets (MPDUs) to this station
+    pub tx_failed: Option<u32>,
+    /// Total transmitted packets (MSDUs and MMPDUs) to this station
+    pub tx_packets: Option<u32>,
+    /// Total retries (MPDUs) to this station
+    pub tx_retries: Option<u32>,
 }
 
-impl Station {
-    pub fn default() -> Station {
-        Station {
-            bssid: None,
-            connected_time: None,
-            beacon_loss: None,
-            signal: None,
-            average_signal: None,
-            rx_packets: None,
-            tx_packets: None,
-            rx_bitrate: None,
-            tx_bitrate: None,
-            tx_retries: None,
-            tx_failed: None,
-        }
-    }
-}
-
-impl ParseNlAttr for Station {
+impl FromNlAttributeHandle for Station {
     /// Parse netlink messages returned by the nl80211 command CmdGetStation
-    fn parse(&mut self, handle: AttrHandle<Nl80211Attr>) -> Station {
+    fn from_handle(handle: AttrHandle<Nl80211Attr>) -> Result<Station, NlError> {
+        let mut station = Station {
+            ..Default::default()
+        };
         for attr in handle.iter() {
             match attr.nla_type {
-                Nl80211Attr::AttrMac => {
-                    self.bssid = Some(attr.payload.clone());
-                }
+                Nl80211Attr::AttrMac => station.bssid = Some(parse_macaddr(&attr.payload)?),
                 Nl80211Attr::AttrStaInfo => {
                     let sub_handle = attr.get_nested_attributes::<Nl80211StaInfo>().unwrap();
```

Comment:
same here, this statement panics if the `get_nested_attributes` call returns in error